exchanges/kucoin/kucoin_futures.go

```diff
@@ -0,0 +1,1087 @@
+		Data bool `json:"data"`
+		Error
+	}{}
+
+	params := make(map[string]interface{})
+	if symbol == "" {
+		return resp.Data, errors.New("symbol can't be empty")
+	}
+	params["symbol"] = symbol
+	params["status"] = status
+	return resp.Data, ku.SendAuthHTTPRequest(ctx, exchange.RestFutures, http.MethodPost, kucoinFuturesSetAutoDeposit, params, &resp)
+}
+
+// AddMargin is used to add margin manually
+func (ku *Kucoin) AddMargin(ctx context.Context, symbol, uniqueID string, margin float64) (*FuturesPosition, error) {
+	resp := struct {
+		Data FuturesPosition `json:"data"`
+		Error
+	}{}
+
+	params := make(map[string]interface{})
+	if symbol == "" {
+		return nil, errors.New("symbol can't be empty")
+	}
+	params["symbol"] = symbol
+	if uniqueID == "" {
+		return nil, errors.New("uniqueID can't be empty")
+	}
+	params["bizNo"] = uniqueID
+	if margin <= 0 {
+		return nil, errors.New("margin can't be zero or negative")
+	}
+	params["margin"] = strconv.FormatFloat(margin, 'f', -1, 64)
+	return &resp.Data, ku.SendAuthHTTPRequest(ctx, exchange.RestFutures, http.MethodPost, kucoinFuturesAddMargin, params, &resp)
+}
+
+// GetFuturesRiskLimitLevel gets information about risk limit level of a specific contract
+func (ku *Kucoin) GetFuturesRiskLimitLevel(ctx context.Context, symbol string) ([]FuturesRiskLimitLevel, error) {
+	resp := struct {
+		Data []FuturesRiskLimitLevel `json:"data"`
+		Error
+	}{}
+
+	return resp.Data, ku.SendAuthHTTPRequest(ctx, exchange.RestFutures, http.MethodGet, fmt.Sprintf(kucoinFuturesRiskLimitLevel, symbol), nil, &resp)
+}
+
+// UpdateRiskLmitLevel is used to adjustment the risk limit level
+func (ku *Kucoin) UpdateRiskLmitLevel(ctx context.Context, symbol string, level int64) (bool, error) {
+	resp := struct {
+		Data bool `json:"data"`
+		Error
+	}{}
+
+	params := make(map[string]interface{})
+	if symbol == "" {
+		return resp.Data, errors.New("symbol can't be empty")
+	}
+	params["symbol"] = symbol
+	params["level"] = strconv.FormatInt(level, 10)
+	return resp.Data, ku.SendAuthHTTPRequest(ctx, exchange.RestFutures, http.MethodPost, kucoinFuturesUpdateRiskLmitLevel, params, &resp)
+}
+
+// GetFuturesFundingHistory gets information about funding history
+func (ku *Kucoin) GetFuturesFundingHistory(ctx context.Context, symbol string, offset, maxCount int64, reverse, forward bool, startAt, endAt time.Time) ([]FuturesFundingHistory, error) {
+	resp := struct {
+		Data struct {
+			DataList []FuturesFundingHistory `json:"dataList"`
+			HasMore  bool                    `json:"hasMore"`
+		} `json:"data"`
+		Error
+	}{}
+
+	params := url.Values{}
+	if symbol == "" {
+		return nil, errors.New("symbol can't be empty")
+	}
+	params.Set("symbol", symbol)
+	if !startAt.IsZero() {
+		params.Set("startAt", strconv.FormatInt(startAt.UnixMilli(), 10))
+	}
+	if !endAt.IsZero() {
+		params.Set("endAt", strconv.FormatInt(endAt.UnixMilli(), 10))
+	}
+	if reverse {
+		params.Set("reverse", "true")
+	} else {
+		params.Set("reverse", "false")
+	}
```

Comment:
in this situation we can probably parse bool just to cut down on code size, this can be replicated across the PR. 

```go
params.Set("reverse", strconv.ParseBool(reverse))
```